extensions/cache/runtime/src/main/java/io/quarkus/cache/runtime/caffeine/CaffeineCacheImpl.java

```diff
@@ -89,6 +89,20 @@ public CompletionStage<V> get() {
                 });
     }
 
+    @Override
+    public <K, V> CompletableFuture<V> getIfPresent(K key) {
```

Comment:
+1 on combining though, but let's keep the generics..

Reply:
I don't have a strong opinion about `V` (unlike `K`) so I'm fine with keeping it. `CaffeineCacheImpl#cast` cannot be used "as is" though because the exception message doesn't make sense for `getIfPresent`. You could cast the value from `getIfPresent` without relying on `CaffeineCacheImpl#cast` at all:
```
caffeineValue.thenApply(value -> (V) value)
```

Reply:
Fix this as well:

* Removed `K`
* Joined methods
* Custom `CacheException` if cast fails